circuit_knitting/utils/simulation.py

```diff
@@ -170,5 +170,6 @@ def _run(
         **run_options,
     ):
         job = PrimitiveJob(self._call, circuits, parameter_values, **run_options)
-        job.submit()
+        # The public submit method was removed in Qiskit 1.0
+        (job.submit if hasattr(job, "submit") else job._submit)()
```

Comment:
Seems a bit odd using a private method here. Then again, you do not really have an alternative with the `PrimitiveJob` interface.

Why exactly is this `ExactSampler` needed? I would assume that this works the same as `Sampler(..., options={"shots": None})`, does it not?

Reply:
We needed to make `ExactSampler` because the reference implementation doesn't handle mid-circuit measurements, which is crucial for us.

Ya, this `(job.submit if ...)()` seemed to be the go-to solution for a couple of other repos as well. The public method didn't carry over from 0.46 to 1.0.

Reply:
This is a good catch though, because they updated the [parent classes in the class signature](https://github.com/Qiskit/qiskit/blob/63c922e617709762837e2b1411d447f90de966b6/qiskit/primitives/sampler.py#L38) to include `PrimitiveJob` when `Sampler` started calling the `PrimitiveJob._submit` private method. We should do the same.